internal/cmd/generator/vm.go.tmpl

```diff
@@ -193,7 +194,8 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 				ifacePtr = iface.ptr
 				typ = iface.typ
 			}
-			if ifacePtr == nil {
+			isDirectedNil := typ != nil && typ.Kind() == reflect.Struct && !runtime.IfaceIndir(typ)
```

Comment:
maybe check this condition in `if ifacePtr == nil {` ?

It will be executed every times then for is not nil

Reply:
nice catch!